sources/api/early-boot-config/src/provider.rs

```diff
@@ -16,6 +16,11 @@ mod vmware;
 #[cfg(bottlerocket_platform = "vmware")]
 pub(crate) use vmware::VmwareDataProvider as Platform;
 
+#[cfg(bottlerocket_platform = "metal")]
+mod metal;
+#[cfg(bottlerocket_platform = "metal")]
+pub(crate) use metal::MetalDataProvider as Platform;
+
```

Comment:
I'd like to see us drop the `aws-dev` special case rules and just treat local file in `/var/lib/bottlerocket` as a valid source for all platforms.

That will streamline the platforms we support to just the three (`aws`, `vmware`, `metal`) rather than "three and a little more."

Reply:
Opened a new issue for this request - I'll take care of it!